test/opt/test_types.cpp

```diff
@@ -218,4 +218,20 @@ TEST(Types, AllTypes) {
   }
 }
 
+#define TestWidthOfType(ty, ...)                                      \
+  TEST(Types, TypeWidthOf##ty) {                                      \
+    std::vector<uint32_t> widths = {1, 2, 4, 8, 16, 32, 48, 64, 128}; \
+    std::vector<std::unique_ptr<ty>> types;                           \
+    for (uint32_t w : widths) {                                       \
+      types.emplace_back(new ty(w, ##__VA_ARGS__));                   \
```

Comment:
http://stackoverflow.com/questions/5588855/standard-alternative-to-gccs-va-args-trick